src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +345,98 @@ def load_secure_props(self) -> None:
                 f" with error '{error_msg}'",
                 SecurePropsNotFoundWarning,
             )
+        # self.set_secure_props() 
+
+    def _retrieve_password(self, service_name: str) -> Optional[str]:
+        """
+        Retrieve the password from the keyring or storage.
+        If the password exceeds the maximum length, retrieve it in parts.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the password retrieval
+        Returns
+        -------
+        str
+            The retrieved password
+        """
+        password = keyring.get_password(service_name, constants["ZoweAccountName"])
+
+        if password is None:
+            # Retrieve the secure value with an index
+            index = 1
+            temp_value = keyring.get_password(service_name, f"{constants['ZoweAccountName']}-{index}")
+            while temp_value is not None:
+                if password is None:
+                    password = temp_value
+                else:
+                    password += temp_value
+                index += 1
+                temp_value = keyring.get_password(service_name, f"{constants['ZoweAccountName']}-{index}")
+                
+        if password is not None and password.endswith("\0"):
+            password = password[:-1]
+
+        return password 
+    
+    def set_secure_props(self) -> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+
+        try:
+            service_name = constants["ZoweServiceName"]
+            credential = self.secure_props.get(self.filepath, {})
+            # Check if credential is a non-empty string
+            if credential:
+                # Get the username and password from the credential dictionary
+                username = credential.get("profiles.base.properties.user")
+                password = credential.get("profiles.base.properties.password")
+                
+                # Combine the username and password as "username:password"
+                username_password = f"{username}:{password}"
+                
+                # Encode the combined string as base64
+                encoded_credential = base64.b64encode(username_password.encode()).decode()
+                if sys.platform == "win32":
+                    service_name += "/" + constants["ZoweAccountName"]
+                    
+                    # Load existing credentials
+                    existing_credential = keyring.get_password(service_name, constants["ZoweAccountName"])
```

Comment:
We should call the `_retrieve_password` helper function here in case the credentials are split across multiple vault entries on Windows.